docs/reference/specifications/providers.md

```diff
@@ -0,0 +1,251 @@
+
+!!! note
+
+    flagd supports the OFREP protocol, meaning client-side OFREP providers can also be used for client-side use-cases.
+
+### RPC Mode Provider Metadata
+
+The provider metadata includes properties returned from the [provider_ready event payload](./protos.md#eventstreamresponse) data.
+
+## In-Process Providers
+
+In-process providers use the [sync schema](./protos.md#syncflagsresponse) to connect to flagd, initiate the [sync stream](./protos.md#eventstreamresponse), and download the `flag-set` rules to evaluate them locally.
+In-process providers are relatively complex (compared to RPC providers) to implement since they essentially must implement more of flagd's logic to evaluate flags locally.
+Local evaluation has the impact of much lower latency and almost no serialization compared to RPC providers.
+
+### JsonLogic Evaluation
+
+An in-process flagd providers provide the feature set offered by [JsonLogic](https://jsonlogic.com) to evaluate flag resolution requests for a given context.
+
+### Custom JsonLogic Evaluators
+
+In addition to the built-in evaluators provided by JsonLogic, the following custom targeting rules are implemented by the provider:
+
+- [Fractional operation](../../reference/custom-operations/fractional-operation.md)
+- [Semantic version evaluation](../../reference/custom-operations/semver-operation.md)
+- [StartsWith/EndsWith evaluation](../../reference/custom-operations/string-comparison-operation.md)
+
+### Targeting Key
+
+Similar to the flagd daemon, in-process providers map the [targeting-key](https://openfeature.dev/specification/glossary#targeting-key) into a top level property of the context used in rules, with the key `"targetingKey"`.
+
+### `$flagd` Properties in the Evaluation Context
+
+Similar to the flagd daemon, in-process flagd providers add the following properties to the JsonLogic evaluation context so that users can use them in their targeting rules.
+Conflicting properties in the context will be overwritten by the values below.
+
+| Property           | Description                                             |
+| ------------------ | ------------------------------------------------------- |
+| `$flagd.flagKey`   | the identifier for the flag being evaluated             |
+| `$flagd.timestamp` | a unix timestamp (in seconds) of the time of evaluation |
+
+### Sync-Metadata Properties in the Evaluation Context
+
+In-process flagd providers also inject any properties returned by the [sync-metadata RPC response](./protos.md#getmetadataresponse) into the context.
+This allows for static properties defined in flagd to be added to in-process evaluations.
+If only a subset of the sync-metadata response is desired to be injected into the evaluation context, you can use the define a mapping function with the `contextEnricher` option.
```

Comment:
We don't yet actually support the addition of arbitrary props into the evaluation context in flagd itself. If this is merged, I will create an issue for that.